### PR TITLE
Change FileName to be relative to project root

### PR DESCRIPTION
**Pull request by SimonCropp**
hopefully helps with #327

### CODE REVIEW

**Comment by SimonCropp** [comment]
@bruno-garcia this embeds the project and solution dir into the assembly that consumes Sentry. 

so where would i wire this in to help with https://github.com/getsentry/sentry-dotnet/issues/327

**Review by bruno-garcia — COMMENTED** [review]
Nice! yeah, I think we need to figure out where that goes into the payload next.
@untitaker do you know about this? (see: https://github.com/getsentry/sentry-dotnet/issues/327)

**Comment by untitaker** [comment]
What is the question?

**Comment by bruno-garcia** [comment]
> What is the question?

We have `abs_path`

```
abs_path
The absolute path to the source file.
```

We currently have the rooted path there `D:\bla\something\else\app\file.cs`.
This PR allows us to know where `app`. 

Do we add the 'project root path' somewhere to the payload? Or do we just truncate the `D:\bla\something\else` out of the `abs_path` and just send the relevant 'project path'? More details on: https://github.com/getsentry/sentry-dotnet/issues/327#issue-541012498

**Comment by untitaker** [comment]
You're correct, you want to put the relative path into filename. That's what the python sdk does at least

sorry for the slow response, i definetly replied via email but for some reason it didn't show up here

File: src/Sentry/buildTransitive/Sentry.props [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        <_Parameter1>Sentry.SolutionDirectory</_Parameter1>
+        <_Parameter2>$(SolutionDir)</_Parameter2>
+      </SentryAttributes>
+      <!-- Ensure not part of Compile, as a workaround for https://github.com/dotnet/sdk/issues/114 -->
+      <Compile Remove="$(SentryAttributesFilePath)" />
+    </ItemGroup>
+    <WriteCodeFragment AssemblyAttributes="@(SentryAttributes)"
+                       Language="$(Language)"
+                       OutputFile="$(SentryAttributesFilePath)">
+      <Output TaskParameter="OutputFile" ItemName="Compile" Condition="$(Language) != 'F#'" />
```
Comment (bruno-garcia) [inline src/Sentry/buildTransitive/Sentry.props:30]: `Condition="$(Language) != 'F#'"` does this mean we want this for VB or a possible IronRuby or something if that's a thing?

File: test/Sentry.Tests/AttributeReaderTests.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using AttributeReader = Sentry.AttributeReader;
+
+public class AttributeReaderTests
+{
+    [Fact]
```
Comment (bruno-garcia) [inline test/Sentry.Tests/AttributeReaderTests.cs:5]: Cool, this should run on iOS AOT'ed so we would see how it behaves there. Assuming the tests are running there following #1703

**Comment by mattjohnsonpint** [comment]
Thats odd.  It seems to be coming up blank on Mono in CI:

```
Error: Assert.Equal() Failure
          ↓ (pos 0)
Expected: Internals/SentryStackTraceFactoryTests.cs
Actual:   
          ↑ (pos 0)
```

But it works on my local machine, even on Mono.  hmmm

**Comment by SimonCropp** [comment]
@mattjohnsonpint  r u running in release mode locally?

**Comment by mattjohnsonpint** [comment]
Testing further, it doesn't matter if debug vs release mode.  It passes on all targets when run from Rider's test runner.  It fails on NET461 (Mono) when run from `dotnet test` on the command line.

I've seen before where I get filenames and line numbers in Rider, but that was when a debugger was attached.  This is in the test runner.  Whatever they are doing special, they're doing it for tests as well.

File: src/Sentry/Sentry.csproj [hunk 3]
```diff
@@ -55,4 +55,9 @@
     <PackageReference Include="System.Text.Json" Version="5.0.2" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Content Include="buildTransitive\Sentry.props" PackagePath="buildTransitive\Sentry.props" />
+    <Content Include="buildTransitive\Sentry.props" PackagePath="build\Sentry.props" />
```
Comment (mattjohnsonpint) [inline src/Sentry/Sentry.csproj:60]: I'm seeing two warnings in the build output related to these:

```
buildTransitive/Sentry.props : warning XA0101: @(Content) build action is not supported [/Users/matt/Code/getsentry/sentry-dotnet/src/Sentry/Sentry.csproj]
buildTransitive/Sentry.props : warning XA0101: @(Content) build action is not supported [/Users/matt/Code/getsentry/sentry-dotnet/src/Sentry/Sentry.csproj]
```

**Comment by mattjohnsonpint** [comment]
FYI - even if I build and test (passing) in Rider, and then test without building so it's using the exact same code (`dotnet test test/Sentry.Tests --filter FileNameShouldBeRelative -f net461 --no-build`), it still fails.  So I don't believe it's a compiler flag, but something being done at runtime.

File: src/Sentry/Sentry.csproj (hunk 3 above)
Comment (SimonCropp) [reply src/Sentry/Sentry.csproj:60]: fixed

File: src/Sentry/buildTransitive/Sentry.props (hunk 1 above)
Comment (SimonCropp) [reply src/Sentry/buildTransitive/Sentry.props:30]: for some reason the item name for F# is `CompileBefore` but for VB and c# it is `Compile`. 

for other languages (like IronRuby) i have not tested. do you want me to include any other languages? Currently they will be a no-op

**Review by mattjohnsonpint — APPROVED** [review]

**Comment by untitaker** [comment]
You're correct, you want to put the relative path into filename. That's
what the python sdk does at least

**Pull request merged**